tests/phpunit/Persistence/Search/Query/RangeFacetQueryBuilderTest.php

```diff
@@ -0,0 +1,132 @@
+<?php
+
+declare( strict_types = 1 );
+
+namespace ProfessionalWiki\WikibaseFacetedSearch\Tests\Persistence\Search\Query;
+
+use Elastica\Query;
+use PHPUnit\Framework\TestCase;
+use ProfessionalWiki\WikibaseFacetedSearch\Application\FacetConfig;
+use ProfessionalWiki\WikibaseFacetedSearch\Application\FacetType;
+use ProfessionalWiki\WikibaseFacetedSearch\Application\PropertyConstraints;
+use ProfessionalWiki\WikibaseFacetedSearch\Persistence\Search\Query\DelegatingFacetQueryBuilder;
+use ProfessionalWiki\WikibaseFacetedSearch\Persistence\Search\Query\RangeFacetQueryBuilder;
+use Wikibase\DataModel\Entity\ItemId;
+use Wikibase\DataModel\Entity\NumericPropertyId;
+use Wikibase\DataModel\Services\Lookup\InMemoryDataTypeLookup;
+
+/**
+ * @covers \ProfessionalWiki\WikibaseFacetedSearch\Persistence\Search\Query\RangeFacetQueryBuilder
+ */
+class RangeFacetQueryBuilderTest extends TestCase {
+
+	private const QUANTITY_PROPERTY = 'P100';
+	private const STRING_PROPERTY = 'P200';
+	private const TIME_PROPERTY = 'P300';
+	private const ITEM_PROPERTY = 'P400';
+
+	public function testBuildsQueryForQuantityRange(): void {
+		$query = $this->newFacetQueryBuilder()->buildQuery(
+			$this->newRangeFacetConfig( self::QUANTITY_PROPERTY ),
+			new PropertyConstraints(
+				new NumericPropertyId( self::QUANTITY_PROPERTY ),
+				min: 42,
+				max: 9001
+			)
+		);
+
+		$this->assertIsRangeQueryWithParams(
+			self::QUANTITY_PROPERTY,
+			[ 'gte' => 42.0, 'lte' => 9001.0 ],
+			$query
+		);
+	}
+
+	private function newFacetQueryBuilder(): RangeFacetQueryBuilder {
+		return new RangeFacetQueryBuilder(
+			$this->newDataTypeLookup()
+		);
+	}
+
+	private function newDataTypeLookup(): InMemoryDataTypeLookup {
+		$dataTypeLookup = new InMemoryDataTypeLookup();
+
+		$types = [
+			self::QUANTITY_PROPERTY => 'quantity',
+			self::STRING_PROPERTY => 'string',
+			self::TIME_PROPERTY => 'time',
+			self::ITEM_PROPERTY => 'wikibase-item'
+		];
+
+		foreach ( $types as $pId => $type ) {
+			$dataTypeLookup->setDataTypeForProperty(
+				new NumericPropertyId( $pId ),
+				$type
+			);
+		}
```

Comment:
I suspect this would be clearer without the loop, but that is a minute CS difference

Reply:
I copied it from WikibaseExport: https://github.com/ProfessionalWiki/WikibaseExport/blob/d3b9f998fff2144e9f1ffd1a48040791917267aa/tests/Application/Export/ProductionValueSetCreatorTest.php#L44-L67

At some point we might need to define more types here: https://github.com/ProfessionalWiki/WikibaseFacetedSearch/issues/58

For future reference: is the concern here about having a loop for few items, or having a loop here in general?